### ISSUE TITLE
Minimap too small on Retina displays

### ISSUE THREAD

**Issue by Crazor** [OpenTTD/OpenTTD#8998]
## Version of OpenTTD
1.11.0

## Expected result
I expect to be able to zoom into the minimap at least one, better two more steps.

## Actual result
The minimap is too small on retina screens even when fully zoomed in

## Steps to reproduce
Run OpenTTD on a Mac with Retina display, open the minimap, zoom in all the way.

## System and Settings
* MacBook Air 2020 13" (2560x1600px)
* System display settings: Select "scaled" and then the rightmost icon for maximum area
* UI element size: quad
* Font size: double
* openttd.cfg:
```
small_font = /Users/dhe/Library/Fonts/DejaVuSans.ttf
medium_font = /Users/dhe/Library/Fonts/DejaVuSans-Bold.ttf
large_font = /Users/dhe/Library/Fonts/DejaVuSerif-Bold.ttf
mono_font = /Users/dhe/Library/Fonts/DejaVuSansMono-Bold.ttf
small_size = 14
medium_size = 15
large_size = 20
mono_size = 14
```

The above settings give me a nice and playable OpenTTD experience. Viewport zoom levels and UI size are perfect:
<img width="1792" alt="Bildschirmfoto 2021-04-10 um 16 12 53" src="https://user-images.githubusercontent.com/128295/114272752-b108c780-9a17-11eb-84ff-6db97d9a1bd5.png">

The minimap is not, however. This is a 128x256 map:
<img width="1792" alt="Bildschirmfoto 2021-04-10 um 16 13 03" src="https://user-images.githubusercontent.com/128295/114272794-d7c6fe00-9a17-11eb-929c-2a1cac670f62.png">

I think this configuration could use at least one, if not two more levels to zoom in. I briefly looked into the source code and noticed that as zoom levels are "increase to zoom out", there is no easy way for me to set a zoom level below 1. Fractional zoom levels seem to need a bunch of work. Maybe the minimap scaling should be always blown up to 2x or 4x size depending on the UI element size setting? The following is a 4096x4096 map zoomed out all the way, so maybe there should be one or two additional zoom-out levels added as well then:
<img width="1792" alt="Bildschirmfoto 2021-04-10 um 16 17 56" src="https://user-images.githubusercontent.com/128295/114272928-6e93ba80-9a18-11eb-946f-4097f48f04cd.png">

**Comment by JGRennison** [comment]
@ldpl was looking into this recently https://github.com/citymania-org/cmclient/commit/6f8e0c144b30afe95e78dab505d9094e520be8a0

I've not actually tried the above myself.

**Comment by Crazor** [comment]
Yup, can confirm that CM's Minimap is looking great on my machine (128x256 map):
<img width="1792" alt="Bildschirmfoto 2021-04-10 um 16 49 24" src="https://user-images.githubusercontent.com/128295/114274019-e1069980-9a1c-11eb-850a-566d7c8459b9.png">

**Comment by Wolfolo** [comment]
I don't have a retina display (or a Mac) but I would like a bigger smallmap too, I usually play 64x64 and I have the same problem shown on the second image, so here is my +1

And maybe I could even PR it by myself if I'll find some will power to do it.

**Comment by Crazor** [comment]
Interesting, I thought that this was an artifact of the doubled up resolution. I used to play a lot a couple of years ago, and from what I remember also very small maps, however I could not remember seeing the tiny minimap before.

So just pixel-doubling for highdpi systems does not seem to be the solution to this issue. Just to be sure (since "retina display" is an Apple marketing term): @Wolfolo, what's your resolution and panel size? It's not one of those 4k displays, is it?

**Comment by Wolfolo** [comment]
The good old 1920x1080 iirc set as windowed matching desktop resolution, so nothing so weird.
I'll retry with different settings

**Comment by Sgt-Nukem** [comment]
I installed the April Fools Day release on Steam just for some quick childhood memories but got quickly addicted again to it. Nice work guys! 💪

But this is one of the main issues I have with it as well.

Here's OpenTTD on a Samsung CRG9 32:9 monitor with its 5120x1440 resolution at double-font-size and double-ui-element-size (which make it a perfect fit otherwise):
![1536610_20210414191756_1](https://user-images.githubusercontent.com/4144388/116921606-7e3ba300-ac54-11eb-9089-33f092526b63.png)
As you can see by the disabled 'plus' icon the minimap is **at maximum zoom level** currently!

It's not just the zoom level that is missing AT LEAST 3 more steps, the minimap window has other issues as well:
- The **font size** for the map legend is **FAAAAAR too small** - it doesn't seem to care about 'double-font-size' at all
- The colored boxes in the map legend are too small, too. There's also some margin between them and their explanation missing. (And some "justify right" for the text showing the height levels in meters.)
- Its the only window I could find where the "right-click to close window" UI override is ignored. (I would rather have it ignored at the freight overview / train details window, where right-clicking an empty wagon could show what KIND of cargo it can carry. No need to click through the tabs then.)
- The rectangle corners that show where the virtual camera is at the moment is just guess work. To be fair this needs also fixing at regular Full HD resolution IMHO, as it is pretty indecipherable. There should maybe be some helpful "a little bit darker" 75% transparent overlay on the area where the camera rectangle IS NOT currently - like some kind of vignette effect - I don't know.
- Because of the rectangle thingie it cannot provide its IMO main function: Showing you the industry that you currently need for a transport line which is nearest to the already existing station. Because even little clicks left and right completely mess with your current in-game postion you have no idea where you are on that minimap window (at least on larger maps).

**Comment by Sgt-Nukem** [comment]
> @ldpl was looking into this recently [citymania-org/cmclient@6f8e0c1](https://github.com/citymania-org/cmclient/commit/6f8e0c144b30afe95e78dab505d9094e520be8a0)
> 
> I've not actually tried the above myself.

But I did. I found this "CityMania" thing some weeks ago. Not knowing at all what "CityMania" is supposed to be - the website doesn't tell much - but it said "100% compatible with OpenTTD" so I downloaded it and used it for a multiplayer match with a friend.

a) Yes. The zoom levels for the minimap are far better than in OpenTTD.
b) It crashed after ~ 1 hour of gameplay. So I quickly switched back to the OpenTTD client for multiplayer.

**Comment by ldpl** [comment]
> b) It crashed after ~ 1 hour of gameplay. So I quickly switched back to the OpenTTD client for multiplayer.

Have you reported that crash anywhere? Like https://github.com/citymania-org/cmclient/issues for example.
Though if it was on 1.11.0 it's likely already fixed.

**Comment by perryprog** [comment]
Any progress on getting a fix for this? The issue is still present on up-to-date versions of macOS for OpenTTD 12.1.

**Comment by 2TallTyler** [comment]
Can someone try this on 13.1? We added interface scaling in 13.0 and I don't have a Mac to test if that resolved this.

**Comment by vidmik** [comment]
Unfortunately it seems like the issue is still there with OpenTTD 13.1 (macOS 13.3.1 Ventura).

**Comment by mrpenguinb** [comment]
On Windows, the map is not affected by the interface scaling. Even on a 1440p the map is mostly useless....

**Comment by Zaravira** [comment]
17" laptop here, Win10 with 1920x1080 resolution, same issue. Map has me crawling into my screen and squinting while the map's fully zoomed in. Tried to play a few years ago but didn't, tried again today and was instantly reminded why I didn't.

Tried that citymania client and the map zoom is beautiful, but I wasn't interested once reading its known bugs list. Please please fix this map zoom!

**Comment by AtFreezingPoint** [comment]
Continuing issue
